rapids-cmake/cpm/detail/gpv_root_dir_hook.cmake

```diff
@@ -0,0 +1,31 @@
+#=============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+# Make sure we always have CMake 3.23 policies when executing this file Since we can be executing in
+# directories of users of rapids-cmake which have a lower minimum cmake version and therefore
+# different policies
+#
+cmake_policy(PUSH)
+cmake_policy(VERSION 3.23)
+
+# Include the needed functions that write out the the pinned versions file
+include("${rapids-cmake-dir}/cpm/detail/gpv_write_file.cmake")
```

Comment:
Could we use a name like `pinning` instead of `gpv`? `pinning_write_file`, `rapids_cpm_pinning_write_file`, etc. are clearer to me.

Alternatively, should we call this a "lockfile" and call the features "write lockfile", etc?

Reply:
I wanted to make it clear what user level function ( `generate_pinned_versions` ) that these two files are related too.
But felt using `generate_pinned_versions` in each file and function signature was too verbose. Maybe we just accept that cost...

Or we go with a shorter word like `pinning` for the detail side. @bdice thoughts?

Reply:
I'm okay with `pinning` for the detail side. That's more meaningful than `gpv`, in my opinion.